sdk/src/main/scala/io/horizen/account/state/ForgerBlockCountersSerializer.scala

```diff
@@ -1,27 +1,46 @@
 package io.horizen.account.state
 
-import io.horizen.account.proposition.{AddressProposition, AddressPropositionSerializer}
+import io.horizen.account.proposition.AddressPropositionSerializer
+import io.horizen.account.utils.ForgerIdentifier
+import io.horizen.proposition.{PublicKey25519PropositionSerializer, VrfPublicKeySerializer}
 import sparkz.core.serialization.SparkzSerializer
 import sparkz.util.serialization.{Reader, Writer}
 
-object ForgerBlockCountersSerializer extends SparkzSerializer[Map[AddressProposition, Long]] {
+object ForgerBlockCountersSerializer extends SparkzSerializer[Map[ForgerIdentifier, Long]] {
 
   private val addressSerializer: AddressPropositionSerializer = AddressPropositionSerializer.getSerializer
+  private val signPubKeySerializer: PublicKey25519PropositionSerializer =
+    PublicKey25519PropositionSerializer.getSerializer
+  private val vrfPubKeySerializer: VrfPublicKeySerializer = VrfPublicKeySerializer.getSerializer
+  final val SERIALIZATION_FORMAT_1_4_FLAG = -1
 
-  override def serialize(forgerBlockCounters: Map[AddressProposition, Long], w: Writer): Unit = {
+  override def serialize(forgerBlockCounters: Map[ForgerIdentifier, Long], w: Writer): Unit = {
     w.putInt(forgerBlockCounters.size)
-    forgerBlockCounters.foreach { case (address, counter) =>
-      addressSerializer.serialize(address, w)
-      w.putLong(counter)
+    forgerBlockCounters.foreach {
+      case (forgerIdentifier, counter) =>
+        addressSerializer.serialize(forgerIdentifier.address, w)
+        if (forgerIdentifier.blockSignPublicKey.isDefined && forgerIdentifier.vrfPublicKey.isDefined) {
```

Comment:
If for any reason we defined just one of the key, that would result in skipping the whole serialization. This is unlikely as of now, but probably it would be better to have one only optional pair.

Reply:
Fixed, will test running against mainnet just for extra measure